contracts/tokenLocking/TokenLocking.sol

```diff
@@ -149,11 +189,64 @@ contract TokenLocking is TokenLockingStorage, DSMath { // ignore-swc-123
     }
   }
 
+  function approveStake(address _user, uint256 _amount, address _token) public calledByColony() {
+    approvals[_user][_token][msg.sender] = add(approvals[_user][_token][msg.sender], _amount);
+  }
+
+  function obligateStake(address _user, uint256 _amount, address _token) public calledByColony() {
+    approvals[_user][_token][msg.sender] = sub(approvals[_user][_token][msg.sender], _amount);
+    obligations[_user][_token][msg.sender] = add(obligations[_user][_token][msg.sender], _amount);
+    totalObligations[_user][_token] = add(totalObligations[_user][_token], _amount);
+
+    require(userLocks[_token][_user].balance >= totalObligations[_user][_token], "colony-token-locking-insufficient-deposit");
+  }
+
+  function deobligateStake(address _user, uint256 _amount, address _token) public calledByColony() {
+    obligations[_user][_token][msg.sender] = sub(obligations[_user][_token][msg.sender], _amount);
+    totalObligations[_user][_token] = sub(totalObligations[_user][_token], _amount);
+  }
+
+  function slashStake(address _user, uint256 _amount, address _token, address _beneficiary) public calledByColony() {
+    obligations[_user][_token][msg.sender] = sub(obligations[_user][_token][msg.sender], _amount);
+    totalObligations[_user][_token] = sub(totalObligations[_user][_token], _amount);
+
+    // Transfer the the tokens
+    Lock storage userLock = userLocks[_token][_user];
+    Lock storage beneficiaryLock = userLocks[_token][_beneficiary];
+    userLock.balance = sub(userLock.balance, _amount);
+    beneficiaryLock.pendingBalance = add(beneficiaryLock.pendingBalance, _amount);
```

Comment:
Can we get the consistent behaviour here too?

Reply:
Sure sure